packages/taco-auth/src/providers/eip4361.ts

```diff
@@ -35,10 +35,17 @@ export class EIP4361AuthProvider {
 
   private async createSIWEAuthMessage(): Promise<AuthSignature> {
     const address = await this.signer.getAddress();
+    const chainId = (await this.provider.getNetwork()).chainId;
     const { domain, uri } = this.getParametersOrDefault();
+    const message = EIP4361AuthProvider.prepareSiweMessage(address, chainId, domain, uri);
+    const signature = await this.signer.signMessage(message);
+    const scheme = 'EIP4361';
+    return { signature, address, scheme, typedData: message };
+  }
+
+  public static prepareSiweMessage(address: string, chainId: number, domain: string, uri: string) {
```

Comment:
Is this static method related to the TODO on L61?

I might be missing the use case for this method? Just a helper for the user to create a SIWE message to then use for the context variable? If so, it seems pretty restrictive since there are more values that SIWE allows apps to customize.

In terms of our default EIP4361 auth provider behaviour, there is a question about what values to use for `domain` and `uri`, and whether the user can customize that, so wondering if this is related?

Reply:
> Just a helper for the user to create a SIWE message to then use for the context variable?

Yes, it's just a helper function to expose a SIWE-message-creating logic to end-users. I think I will also refactor-out the types related to SIWE so that it's more configurable and gives users more freedom (as you've noted).

> so wondering if this is related?

Yes, it's related, but I planned on fixing the defaults later. I will attempt to squeeze it into this PR, but if it doesn't make sense I will follow up in the next one.

Reply:
> Yes, it's just a helper function to expose a SIWE-message-creating logic to end-users. I think I will also refactor-out the types related to SIWE so that it's more configurable and gives users more freedom (as you've noted).

In that case, I'm a bit apprehensive about the benefits of us trying to provide extensive SIWE-message-creating logic. Then we just end up either making something more restrictive (less parameters than the SIWE API) or writing a layer on top of the SIWE library that uses the same API parameterization of the library anyway. Users should just use the `siwe` library or some other one for generating the SIWE message if they want such a high level of customization of messages.

At the moment, we have two worlds:
1. Apps that expect TACo to do the SIWE info for them because they don't use SIWE at all (`:userAddress`)
2. Apps that currently use SIWE and expect to provide TACo with their existing SIWE info so that it can be reused (`:userAddressExternalEIP4361`)

In world 1) apps don't currently use SIWE at all, so TACo will create/generate the AuthSignature on their behalf. At most, the app may need to specify the `domain` / `uri` if `taco-auth` can't detect those values from the `window` object / use some default value to get the right value based on the SIWE spec (i.e. prevent the signing from being flagged as "unsafe"). But these apps probably don't need more customization than that, or else they would probably be in world 2) anyway.

In world 2) the app uses their own tooling for the SIWE message/signature and all they want to do is give that information to TACo (Ceramic example). No customization of SIWE message parameters needed here because the message and signature are already generated by the app separately.

Reply:
I see your point. I had a slightly different idea of how to go about this, but I see merit in this solution. 

I dropped the commit with previous changes and added an ability to set defaults for the EIP4361 provider. I will expose it as a part of https://github.com/nucypher/taco-web/issues/533

Reply:
> I had a slightly different idea of how to go about this, but I see merit in this solution.

Happy to listen to your ideas.